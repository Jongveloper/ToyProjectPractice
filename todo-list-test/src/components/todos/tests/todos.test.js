import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Todos from '../todos';

describe('Todos component', () => {
  const todos = [
    { name: 'Learning React', id: 1 },
    { name: 'Learning Nodejs', id: 2 },
  ];
  let TodoComponent;
  let onDelete;
  let onAdd;

  beforeEach(() => {
    onAdd = jest.fn();
    onDelete = jest.fn();
    TodoComponent = <Todos todos={todos} onDelete={onDelete} onAdd={onAdd} />;
  });

  it('renders', () => {
    const component = renderer.create(TodoComponent);
    expect(component.toJSON()).toMatchSnapshot();
  });

  const setup = () => render(TodoComponent);

  it('할 일을 등록합니다.', () => {
    setup();
    const input = screen.getByPlaceholderText('해야할 일을 입력하세요');
    const button = screen.getByText('Add');
    const newTodo = 'Learning nextjs';
    userEvent.type(input, newTodo);
    userEvent.click(button);
    expect(onAdd).toHaveBeenCalledWith(newTodo);
  });

  it('todo를 삭제합니다.', () => {
    setup();
    const button = screen.getByTitle('delete')[0];
    userEvent.click(button);
    expect(onDelete).toHaveBeenCalledWith(todos[0]);
  });
});
