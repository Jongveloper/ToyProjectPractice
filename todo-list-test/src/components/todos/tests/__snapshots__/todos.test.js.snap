// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos component renders 1`] = `
<div>
  <form
    className="add-from"
    onSubmit={[Function]}
  >
    <input
      className="add-input"
      placeholder="해야할 일을 입력하세요"
      type="text"
    />
    <button
      className="add-button"
    >
      Add
    </button>
  </form>
  <ul>
    <li>
      <span />
      <button
        onClick={[Function]}
        title="삭제"
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </li>
    <li>
      <span />
      <button
        onClick={[Function]}
        title="삭제"
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </li>
  </ul>
</div>
`;
